core/lib/snapshots_applier/src/lib.rs

```diff
@@ -0,0 +1,282 @@
+
+    async fn fetch_newest_snapshot(&self) -> Result<Option<SnapshotHeader>, RpcError>;
+}
+impl<'a> SnapshotsApplier<'a> {
+    pub async fn load_snapshot(
+        connection_pool: &'a ConnectionPool,
+        main_node_client: Box<dyn SnapshotsApplierMainNodeClient>,
+        blob_store: Box<dyn ObjectStore>,
+    ) -> anyhow::Result<(), SnapshotApplierError> {
+        let mut storage = connection_pool
+            .access_storage_tagged("snapshots_applier")
+            .await?;
+        let mut storage = storage.start_transaction().await?;
+
+        let mut applied_snapshot_status = storage
+            .snapshot_recovery_dal()
+            .get_applied_snapshot_status()
+            .await
+            .map_err(|_| Retryable("Unable to get applied snapshots status".to_string()))?;
+
+        if !storage
+            .blocks_dal()
+            .is_genesis_needed()
+            .await
+            .map_err(|_| Retryable("Unable to fetch genesis state".to_string()))?
+            && applied_snapshot_status.is_none()
+        {
+            return Err(Canceled(
+                "This node has already been initialized without a snapshot".to_string(),
+            ));
+        }
+
+        if let Some(applied_snapshot_status) = applied_snapshot_status.as_ref() {
+            if applied_snapshot_status
+                .storage_logs_chunks_ids_to_process
+                .is_empty()
+            {
+                return Err(Canceled(
+                    "This node has already been initialized from a snapshot".to_string(),
+                ));
+            }
+        } else {
+            applied_snapshot_status =
+                Some(SnapshotsApplier::create_fresh_recovery_status(main_node_client).await?)
+        }
+
+        let mut recovery = Self {
+            connection_pool,
+            blob_store,
+            applied_snapshot_status: applied_snapshot_status.unwrap(),
+        };
+
+        recovery.recover_factory_deps(&mut storage).await?;
+
+        storage
+            .snapshot_recovery_dal()
+            .set_applied_snapshot_status(&recovery.applied_snapshot_status)
+            .await
+            .map_err(|err| Fatal(err.into()))?;
+
+        storage.commit().await.map_err(|err| Fatal(err.into()))?;
+
+        recovery.recover_storage_logs().await?;
+
+        Ok(())
+    }
+
+    async fn create_fresh_recovery_status(
+        main_node_client: Box<dyn SnapshotsApplierMainNodeClient>,
+    ) -> Result<SnapshotRecoveryStatus, SnapshotApplierError> {
+        let snapshot_response = main_node_client
+            .fetch_newest_snapshot()
+            .await
+            .map_err(|_| Retryable("Unable to fetch newest snapshot from main node".to_string()))?;
+        if snapshot_response.is_none() {
+            return Err(Canceled("Main node does not have any ready snapshots, skipping initialization from snapshot!".to_string()));
+        }
+        let snapshot = snapshot_response.unwrap();
```

Comment:
Nit: It's idiomatic to use `ok_or` / `ok_or_else` to map `Option::None` to a `Result::Err`.